### ISSUE TITLE
Always FEM context is used by id_() function of ReducedBasisSpace  

### ISSUE THREAD

**Issue by StephaneVeys** [feelpp/feelpp#208]
To evaluate an expression on a set of points we can use evaluateFromContext(_context=ctx,_expr=expr);

Here ctx can be a RB context (RBctx) or a FEM context (FEMctx).

The problem is that when the expression contains an evaluation -- for example idv(u) -- and we call  evaluateFromContext with a RB context, like :
evaluateFromContext( _context=RBctx , _expr=idv(u) );
Then, even if in the function evaluateFromContext the context used is RBctx, when the code is in idv_( context , ...) function of Element class of ReducedBasisSpace, the context used is always FEMctx.

In other words when we write :
ctxrb_type const\* rb_context = dynamic_cast< ctxrb_type const\* >( &context );
The test :
if( rb_context == 0 )  
returns always true, so the code considers that we are always using a FEMctx.

Typically the FEMctx is used to evaluate the expression on a lot of points and RBctx is used to evaluate the expression on a few number of points. So results are exact with FEMctx but the efficiency of the evaluation is not optimal.
